### ISSUE TITLE
Inconsistent focus ring styling 

### ISSUE THREAD

**Issue by LMS007** [hypothesis/client#2226]
_update_

See slack thread
https://hypothes-is.slack.com/archives/C07NXBDNW/p1608060744150100

Doc to outline issue and some proposals
https://docs.google.com/document/d/1iO4uIxp7ECXZnZjnsXQqyxikQKxdtTuZrsbeRNcFba0/edit

I think as a group we are leaning toward # 2.

We need to decide what to do (if anything) and then how to do it. It should also be extended across repositories likey using the new frontend shared package (TBD)

===========================

We currently don't set the focus ring styling for any elements except `<input>` and `<textarea>`

![Screen Shot 2020-06-03 at 11 25 42 AM](https://user-images.githubusercontent.com/3939074/83674888-015c0100-a58d-11ea-9ade-b611d12e423d.png)

This is not too bad, but it only applies to these elements. It's worth noting that we don't add a focus style for any other elements when clicked on via mouse. These two elements however do get focus styling because it is assumed they will take keyboard input and so focus is necessary.

When navigating via keyboard (tab or arrow keys when appropriate), we should also render a focus ring on all focusable elements. We currently default to the browser built in version of this by not overriding the default styling via `:focus-visible` selector via mixin https://github.com/hypothesis/client/blob/master/src/styles/mixins/focus.scss

**Issues:**

1. The default does not match our input / textarea styling
2. The defaults differs from browser to browser and even OS
3. Some elements don't render the focus ring very well and parts of it get cut off, especially when the elements are positions in special ways or overflow is hidden. In such cases adding some padding could help or other css tricks.
4. Other front end apps should get same treatment (LMS, etc) 

The defaults for each browser look slightly different. Firefox is probably the most unappealing as the box-model ring size seems to not follow the others. It also has the least affordance being just a dotted gray outline. Chrome and Safari look about the same and are reasonable defaults. Could we override all of them for something a bit better? Chrome/Edge in Windows 10 has a different color than  Chrome on Mac -- I don't know why yet.

Additionally, would it may be useful in the future to allow a user preference (similar to the highlight color) to change the focus ring color. Its likely that some vision impaired users would not prefer the Safari or Chrome blue or Windows black.

**Firefox v77**
![ff_focus](https://user-images.githubusercontent.com/3939074/83678771-1b004700-a593-11ea-8a68-551ae2008d86.gif)

**Safari 13**
![safari_focus](https://user-images.githubusercontent.com/3939074/83678783-218ebe80-a593-11ea-8baf-157c49646de6.gif)

**Chrome 83**
![chrome_focus](https://user-images.githubusercontent.com/3939074/83678862-3f5c2380-a593-11ea-9b80-417d7ce94ab8.gif)

**Chrome / Edge on Windows 10** 

![Screenshot_1](https://user-images.githubusercontent.com/3939074/83679297-05d7e800-a594-11ea-906f-c4a16f92acc7.png)

**Comment by esanzgar** [comment]
Firefox nightly displays the focus ring very similar to Chrome and Safari.

![image](https://user-images.githubusercontent.com/8555781/95331620-8c863400-08aa-11eb-97b9-0a014748d131.png)

**Comment by esanzgar** [comment]
[Another, more limited, proposal](https://docs.google.com/document/d/1h-2Xg1pjkmV64k65LMCnH6T00bbORU4TBq-d1SCO7yI/edit?usp=sharing).
